### ISSUE TITLE
Woocommerce v6.3.1 order object->customer_id is missing

### ISSUE THREAD

**Issue by jessebrightman** [woothemes/woocommerce#32319]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Woocommerce v6.3.1 order object->customer_id is missing

### Expected behavior

be able to retrieve the customer_id field from the $order object

### Actual behavior

$order->customer_id is blank

previous version of Woo v6.2.2 works as expected

### Steps to reproduce

1.  Go into files
2. Find an order object (ie. in admin-new-order.php)
3. var_dump($order->customer_id)
4. view results and see it's null

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://verlota.com
Site address (URL): https://verlota.com
WC Version: 6.3.1
REST API Version: ✔ 6.3.1
WC Blocks Version: ✔ 6.9.0
Action Scheduler Version: ✔ 3.4.0
WC Admin Version: ✔ 3.2.1
Log Directory Writable: ✔
WP Version: 5.9.2
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 7.4.28
PHP Post Max Size: 50 MB
PHP Time Limit: 60
PHP Max Input Vars: 1000
cURL Version: 7.81.0
OpenSSL/1.1.1n

SUHOSIN Installed: –
MySQL Version: 8.0.28
Max Upload Size: 50 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 6.3.1
WC Database Prefix: wp_
Total Database Size: 1892.12MB
Database Data Size: 1814.81MB
Database Index Size: 77.31MB
wp_woocommerce_sessions: Data: 2.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.16MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 1.52MB + Index: 1.94MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 4.02MB + Index: 2.55MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 3.02MB + Index: 1.52MB + Engine InnoDB
wp_affiliate_wp_affiliatemeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_affiliate_wp_affiliates: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_affiliate_wp_campaigns: Data: 0.00MB + Index: 0.00MB + Engine
wp_affiliate_wp_creatives: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_affiliate_wp_customermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_affiliate_wp_customers: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_affiliate_wp_payouts: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_affiliate_wp_referrals: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_affiliate_wp_rest_consumers: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_affiliate_wp_visits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_ajaxsearchpro: Data: 0.08MB + Index: 0.00MB + Engine InnoDB
wp_ajaxsearchpro_priorities: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_ajaxsearchpro_statistics: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_alm: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_asp_index: Data: 19.56MB + Index: 26.09MB + Engine InnoDB
wp_asp_synonyms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_aws_cache: Data: 1.52MB + Index: 0.00MB + Engine InnoDB
wp_aws_index: Data: 2.03MB + Index: 1.38MB + Engine InnoDB
wp_commentmeta: Data: 0.09MB + Index: 0.13MB + Engine InnoDB
wp_comments: Data: 1.52MB + Index: 0.58MB + Engine InnoDB
wp_hustle_entries: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_hustle_entries_meta: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_hustle_modules: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_hustle_modules_meta: Data: 1.52MB + Index: 0.03MB + Engine InnoDB
wp_hustle_tracking: Data: 0.06MB + Index: 0.08MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_litespeed_avatar: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_litespeed_cssjs: Data: 1326.52MB + Index: 3.03MB + Engine InnoDB
wp_options: Data: 28.39MB + Index: 19.42MB + Engine InnoDB
wp_postmeta: Data: 13.52MB + Index: 6.03MB + Engine InnoDB
wp_posts: Data: 313.52MB + Index: 1.38MB + Engine InnoDB
wp_rafflepress_contestants: Data: 0.08MB + Index: 0.00MB + Engine InnoDB
wp_rafflepress_entries: Data: 0.23MB + Index: 0.00MB + Engine InnoDB
wp_rafflepress_giveaways: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_rafflepress_polls: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_rs_expiredpoints_email: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_rs_templates_email: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_rsgiftvoucher: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_rspointexpiry: Data: 0.22MB + Index: 0.00MB + Engine InnoDB
wp_rsrecordpoints: Data: 0.28MB + Index: 0.00MB + Engine InnoDB
wp_sumo_reward_encashing_submitted_data: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_sumo_reward_send_point_submitted_data: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_swift_performance_warmup: Data: 0.09MB + Index: 0.13MB + Engine InnoDB
wp_term_relationships: Data: 0.09MB + Index: 0.08MB + Engine InnoDB
wp_term_taxonomy: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_tm_taskmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_tm_tasks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_usermeta: Data: 5.52MB + Index: 7.03MB + Engine InnoDB
wp_users: Data: 0.34MB + Index: 0.30MB + Engine InnoDB
wp_vtprd_purchase_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_vtprd_purchase_log_product: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_vtprd_purchase_log_product_rule: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_vtprd_transient_cart_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.09MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.08MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfblockediplog: Data: 0.23MB + Index: 0.00MB + Engine InnoDB
wp_wfblocks7: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfconfig: Data: 2.52MB + Index: 0.00MB + Engine InnoDB
wp_wfcrawlers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wffilechanges: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wffilemods: Data: 55.64MB + Index: 0.00MB + Engine InnoDB
wp_wfhits: Data: 1.02MB + Index: 0.22MB + Engine InnoDB
wp_wfhoover: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfissues: Data: 0.05MB + Index: 0.06MB + Engine InnoDB
wp_wfknownfilelist: Data: 21.55MB + Index: 0.00MB + Engine InnoDB
wp_wflivetraffichuman: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wflocs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wflogins: Data: 0.42MB + Index: 0.16MB + Engine InnoDB
wp_wfls_2fa_secrets: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wfls_settings: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfnotifications: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfpendingissues: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wfreversecache: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wfsnipcache: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wfstatus: Data: 1.03MB + Index: 0.11MB + Engine InnoDB
wp_wftrafficrates: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_bundled_itemmeta: Data: 0.08MB + Index: 0.09MB + Engine InnoDB
wp_woocommerce_bundled_items: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_exported_csv_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_square_customers: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_debug_events: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_tasks_meta: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wss_log: Data: 0.08MB + Index: 0.00MB + Engine InnoDB
wp_wt_iew_action_history: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wt_iew_cron: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wt_iew_ftp: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wt_iew_mapping_template: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_yoast_indexable: Data: 2.52MB + Index: 0.91MB + Engine InnoDB
wp_yoast_indexable_hierarchy: Data: 0.17MB + Index: 0.22MB + Engine InnoDB
wp_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_primary_term: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_yoast_seo_links: Data: 1.52MB + Index: 2.03MB + Engine InnoDB

### Post Type Counts ###

attachment: 1330
custom_css: 2
et_body_layout: 26
et_footer_layout: 5
et_header_layout: 8
et_pb_layout: 123
et_template: 33
et_theme_builder: 1
foobar_notification: 6
jb_store_css: 7
nav_menu_item: 38
oembed_cache: 20
page: 157
popup: 2
popup_theme: 8
post: 202
product: 87
product_variation: 12
pw_bogo: 1
racemailtemplate: 1
revision: 5613
shop_coupon: 30
shop_order: 398
shop_subscription: 11

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (38) ###

User Menus: by Code Atlantic – 1.2.9
ACL Floating Cart for WooCommerce: by AmaderCode Lab – 0.9
Advanced noCaptcha & invisible Captcha: by WP White Security – 6.1.7
Advanced Woo Search: by ILLID – 2.50
AffiliateWP: by AffiliateWP
LLC – 2.2.14

Ajax Search Pro: by Ernest Marcinko – 4.19.4
Divi Product Carousel: by DiviGear – 2.0.0
Google Tag Manager for Wordpress: by Thomas Geiger – 1.14.2
Hustle Pro: by WPMU DEV – 4.4.6
Lock User Account: by teknigar – 1.0.4
Mautic WooCommerce Marketing Automation: by MakeWebBetter – 2.2.3
Product Bundles - Min/Max Items: by SomewhereWarm – 1.4.3
Quiz Results: by  –
Recover Abandoned Cart: by Fantastic Plugins – 22.0
RafflePress Pro: by RafflePress – 1.5.1
SUMO Reward Points: by Fantastic Plugins – 25.1
Sales Reports: by Jesse Brightman – 1.0.0
ShipBob Express Rates: by ShipBob
Inc. – 1.8.0

SS WooCommerce Myaccount Ajax Tabs: by SaurabhSharma – 2.1.0
WebP Express: by Bjørn Rosell – 0.25.0
MultiStep Checkout for WooCommerce: by ThemeHigh – 2.0.6
WooCommerce Order Test: by WP Fix It – 1.6
Woocommerce Sidebar Category Accordion: by Marc Randall – 1.1.3
WooCommerce Chained Products: by StoreApps – 2.9.1
WooCommerce Cost of Goods: by SkyVerge – 2.9.1
WooCommerce Authorize.Net Gateway: by SkyVerge – 3.6.0
WooCommerce Product Bundles: by SomewhereWarm – 6.3.0
WooCommerce Shipment Tracking: by WooCommerce – 1.6.14
Simple Registration for WooCommerce: by Astoundify – 1.5.3
WooCommerce: by Automattic – 6.3.1
Yoast SEO: by Team Yoast – 18.4.1
WP Mail SMTP: by WPForms – 3.3.0
WP Mautic: by Mautic community – 2.4.2
WP-Optimize - Clean, Compress, Cache: by David Anderson
Ruhani Rabin
Team Updraft – 3.2.3

WP Rollback: by Impress.org – 1.7.1
WP Sitemap Page: by Tony Archambeau – 1.9.1
WP User Avatar: by ProfilePress – 2.2.16
Yoast SEO: WooCommerce: by Team Yoast – 14.3

### Inactive Plugins (18) ###

Ajax Load More: by Darren Cooney – 5.5.2
Asset CleanUp Pro: Page Speed Booster: by Gabe Livan – 1.1.9.5
Better Search Replace: by Delicious Brains – 1.3.4
Hustle: by WPMU DEV – 7.4.13.1
Integration with Mautic for WooCommerce - Open Source Marketing Automation: by MakeWebBetter – 2.2.3
Jetpack Boost: by Automattic – 1.4.0
Lazy Load - Optimize Images: by WP Rocket – 2.3.4
LiteSpeed Cache: by LiteSpeed Technologies – 4.5.0.1
Maintenance: by WebFactory Ltd – 4.05
PW WooCommerce BOGO Pro: by Pimwick
LLC – 2.141

Swift Performance Lite: by SWTE – 2.3.6.5
Verlota Shipbob Rates: by WooThemes – 1.0.0
WooCommerce All Products For Subscriptions: by SomewhereWarm – 3.1.15
WooCommerce Stock Synchronization: by Pronamic – 2.5.0
WooCommerce Subscriptions: by WooCommerce – 3.0.5
Wordfence Security: by Wordfence – 7.5.9
WordPress Importer: by wordpressdotorg – 0.7
WP Downgrade | Specific Core Version: by Reisetiger – 1.2.3

### Dropin Plugins (1) ###

maintenance.php: maintenance.php

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: bundle (bundle)
external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #229969 - /shop/
Cart: #229970 - /cart/
Checkout: #229971 - /checkout/
My account: #229972 - /my-account/
Terms and conditions: #207413 - /terms-and-conditions/

### Theme ###

Name: Divi Child
Version: 1.0.0
Author URL: https://verlota.com
Child Theme: ✔
Parent Theme Name: Divi
Parent Theme Version: 4.15.1
Parent Theme Author URL: http://www.elegantthemes.com
WooCommerce Support: ✔

### Templates ###

Overrides: divi-child/woocommerce/cart/cart.php
divi-child/woocommerce/cart/proceed-to-checkout-button.php
divi-child/woocommerce/checkout/form-checkout.php
divi-child/woocommerce/checkout/payment.php
divi-child/woocommerce/checkout/thankyou.php
divi-child/woocommerce/loop/add-to-cart.php
divi-child/woocommerce/loop/rating.php
divi-child/woocommerce/myaccount/form-login.php

### Product Bundles ###

Database Version: 6.3.0
Loopback Test: ✔
Template Overrides: –

### Authorize.Net Credit Card ###

Environment: Production
Tokenization Enabled: ✔
Debug Mode: Off
Webhooks Enabled: ❌
						Not generated

### Action Scheduler ###

Complete: 8,255
Oldest: 2022-02-22 07:36:43 -0800
Newest: 2022-03-22 08:21:35 -0700

Pending: 1
Oldest: 2022-03-22 08:26:35 -0700
Newest: 2022-03-22 08:26:35 -0700

### Status report information ###

Generated at: 2022-03-22 08:22:30 -07:00
`

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by roykho** [comment]
Hi @jessebrightman 

Thanks for reporting however I see you did not perform all the self checks -- more specifically this one ` I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.`  and `This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/). Can you please perform that test and report back if that resolves your issue? Thanks!

**Comment by Lilrock** [comment]
I'm actually having the same issue. Although, the situation I can replicate more accurately.
Multiple of my sites that I work with are varying ranges of versions including the latest builds.

In my situation, the normal customer_id when a created user is created is included in the Order information.
The problem I have is when some of my sites forget to check the option for "Allow guests to checkout". My plugin doesn't allow users to anonymously purchase and causes problem with my entire system.
So, when this happens, they kindly click this option back on and create an account AFTER the order is submitted and when the order is assigned a new user, the customer_id is still null from being a "guest" account.

I would suggest finding this specific situation in WooCommerce code and fix it... I would look personally, but I have my time filled up and I'm not fluent enough on the WooCommerce code to find it quickly.

Please fix so that my bosses can have happier customers (and off my back about this specific issue).

Thank you for all your hard work guys, keep it up. I love the system you've built and it helps millions of people,

Lilrock

**Comment by adamcollingburn** [comment]
@jessebrightman Did you manage to fix this? Mine is doing the same thing as of the last update

**Comment by Lilrock** [comment]
I'm half tempted to recreate this issue in a new report as this issue doesn't represent a recreation of the issue I'm having.
@adamcollingburn Do you have the same issue as @jessebrightman or can you recreate it using my problem I posted earlier? Maybe we should attempt to wake this issue up somehow

**Comment by Lilrock** [comment]
I found out another possible problem. It's quite strange.

I disabled the chance of having guests purchase from the site by turning the option off in the Woo Commerce Accounts settings. However, this did not solve the problem.

If an order has been created and you do one of the following:
- Update information on the customer (phone number was the cause. Maybe anything to do with shipping and billing)
- Change the customer on the order to another account

Than the issue pops up with "Customer_id is empty".

I thought for sure it was just because we had guest purchases, but that now doesn't seem like the case. It seems like the customer_id is lost when the customer data changes from the order.

Since I don't touch any "processed" items, I didn't think to go back to see if any of the customer_ids from the older items are also affected or not.

I have a simple plugin that grabs the order_id, customer_id and order's date. I use this information as a third party software to connect with a Point of Sale system, but when the customer_id is missing, I can't connect the order to the right customer. But, once the order has been processed, I don't go back to the order to do anything.

Hopefully this helps in determining the problem.

I'm currently using Verions 6.6.0, if it weren't for the security issues and back doors, I'd be going back to before 6.3.1 to have a functional customer_id system.

Thanks
